File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/OrderDto.java
package com.sudo248.orderservice.controller.order.dto;

import com.sudo248.orderservice.controller.payment.dto.PaymentInfoDto;
import com.sudo248.orderservice.repository.entity.order.OrderStatus;
import com.sudo248.orderservice.repository.entity.order.Shipment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderDto {
    private String orderId;
    private String cartId;
    private PaymentInfoDto payment;
    private PromotionDto promotion;
    private UserDto user;
    private OrderStatus status;
    private String address;
    private Double totalPrice, totalPromotionPrice, finalPrice, totalShipmentPrice;
    private List<OrderSupplierDto> orderSuppliers;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/order/OrderDto.kt
package com.sudo248.sudoo.data.dto.order

import com.sudo248.sudoo.data.dto.promotion.PromotionDto
import com.sudo248.sudoo.data.dto.user.UserDto
import com.sudo248.sudoo.domain.entity.payment.PaymentInfo
import com.sudo248.sudoo.domain.entity.promotion.PromotionInfo
import java.time.LocalDateTime

data class OrderDto(
    val orderId: String,
    val cartId: String,
    val payment: PaymentInfo? = null,
    val promotion: PromotionInfo? = null,
    val user: UserDto,
    val totalPrice: Double,
    val totalPromotionPrice: Double,
    val totalShipmentPrice: Double,
    val finalPrice: Double,
    val address: String,
    val createdAt: LocalDateTime,
    val orderSuppliers: List<OrderSupplierDto>
)